### PR TITLE
feat(core): add jenkins artifact type

### PR DESCRIPTION
**Pull request by Jammy-Louie**
spinnaker/spinnaker#4101

### CODE REVIEW

File: clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/jenkins/JenkinsArtifactAccount.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+package com.netflix.spinnaker.clouddriver.artifacts.jenkins;
+
+import com.netflix.spinnaker.clouddriver.artifacts.config.ArtifactAccount;
+import com.netflix.spinnaker.clouddriver.artifacts.config.BasicAuth;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
```
Comment (jkschneider) [inline clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/jenkins/JenkinsArtifactAccount.java]: Final fields and `@RequiredArgsConstructor` where possible?

File: clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/jenkins/JenkinsArtifactCredentials.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    super(okHttpClient, account);
+    this.jenkinsArtifactAccount = account;
+    this.name = account.getName();
+  }
+
+  @Override
+  protected HttpUrl getDownloadUrl(Artifact artifact) {
+    String formattedJenkinsAddress = jenkinsArtifactAccount.getAddress().endsWith("/") ? jenkinsArtifactAccount.getAddress() : jenkinsArtifactAccount.getAddress() + "/";
+    String formattedReference = artifact.getReference().startsWith("/") ? artifact.getReference() : "/" + artifact.getReference();
+    String buildUrl = formattedJenkinsAddress + "job/" + artifact.getName() + "/" + artifact.getVersion() + "/artifact" + formattedReference;
```
Comment (jkschneider) [inline clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/jenkins/JenkinsArtifactCredentials.java]: I believe the address is different if version == `lastSuccessfulBuild`?

Comment (Jammy-Louie) [reply clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/jenkins/JenkinsArtifactCredentials.java]: should be fine.
sample output from `lastSuccessfulBuild `
http://cf-spinnaker.ci.cf-app.com/job/Multi-foundation-metrics/lastSuccessfulBuild/artifact/build/libs/multifoundationmetrics-0.3.0-dev.3+b02ecb9.jar

sample output from build `42`
http://cf-spinnaker.ci.cf-app.com/job/Multi-foundation-metrics/42/artifact/build/libs/

File: clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/jenkins/JenkinsArtifactAccount.java (hunk 1 above)
Comment (Jammy-Louie) [reply clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/jenkins/JenkinsArtifactAccount.java]: 👍

**Pull request merged**